File: 2nd_lab/3rd_task/3rd task.py
import math

# Функция для проверки простоты числа
def is_prime(num):
    if num <= 1:
        return False
    for i in range(2, math.isqrt(num) + 1):
        if num % i == 0:
            return False
    return True

# Функция для подсчета количества простых чисел
def count_primes():
    count = 0
    numbers = input("Введите числа через пробел: ").split()
    for num_str in numbers:
        num = int(num_str)
        if is_prime(num):
            count += 1
    return count

# Ввод количества чисел и вызов функции для подсчета простых чисел
result = count_primes()
print("Количество простых чисел:", result)

File: 2nd_lab/3rd_task/3rd task.go
package main

import (
	"fmt"
)

// Функция для проверки простоты числа
func isPrime(num int) bool {
	if num <= 1 {
		return false
	}
	for i := 2; i*i <= num; i++ {
		if num%i == 0 {
			return false
		}
	}
	return true
}

// Функция для подсчета количества простых чисел
func countPrimes(n int) int {
	count := 0
	fmt.Print("Введите число через пробел: ")
	for i := 0; i < n; i++ {
		var num int
		fmt.Scan(&num)
		if isPrime(num) {
			count++
		}
	}
	return count
}

func main() {
	var n int
	fmt.Print("Введите количество чисел: ")
	fmt.Scan(&n)

	result := countPrimes(n)
	fmt.Println("Количество простых чисел:", result)
}
